cleanlab/experimental/datalab/issue_manager.py

```diff
@@ -423,3 +425,136 @@ def verbosity_levels(self) -> Dict[int, Any]:
             1: {"summary": ["num_ood_issues"], "issue": ["nearest_neighbour"]},
             2: {"issue": ["distance_to_nearest_neighbour"]},
         }
+
+
+class NearDuplicateIssueManager(IssueManager):
+    """Manages issues realted to near-duplicate examples."""
+
+    issue_name: str = "near_duplicate"
+
+    def __init__(
+        self,
+        datalab: Datalab,
+        metric: Optional[str] = "cosine",
+        threshold: Optional[float] = None,
+        k: Optional[int] = 10,
+        **_,
+    ):
+        super().__init__(datalab)
+        self.metric = metric
+        self.threshold = threshold
+        self.k = k
+        self.knn = None
+        self.near_duplicate_sets: List[List[int]] = []
+
+    def find_issues(
+        self,
+        features: List[str],
+        **_,
+    ) -> None:
+
+        feature_array = self._extract_embeddings(features)
+        if self.knn is None:
+            self.knn = NearestNeighbors(n_neighbors=self.k, metric=self.metric)
+
+        try:
+            check_is_fitted(self.knn)
+        except:
+            self.knn.fit(feature_array)
+
+        scores, distances = self._score_features(feature_array)
+        self.radius, self.threshold = self._compute_threshold_and_radius()
+
+        self.issues = pd.DataFrame(
+            {
+                f"is_{self.issue_name}_issue": scores < self.threshold,
+                self.issue_score_key: scores,
+            },
+        )
+
+        indices = self.knn.radius_neighbors(feature_array, self.radius, return_distance=False)
+        self.near_duplicate_sets = [
+            duplicates[duplicates != idx] for idx, duplicates in enumerate(indices)
+        ]
+        self.distances = distances
+
+        self.summary = self.get_summary(score=scores.mean())
+        self.info = self.collect_info()
+
+    def collect_info(self) -> dict:
+        issues_dict = {
+            "num_near_duplicate_issues": len(self.near_duplicate_sets),
```

Comment:
Currently, this is just the number of examples in the dataset.
Should `"num_near_duplicate_issues"` be:

1. Total number of examples with near duplicates?
2. Total number of near duplicates?